OpenSim/Sandbox/MatlabScripts/createActuatorsFile.m

```diff
@@ -0,0 +1,156 @@
+% permissions and limitations under the License.                          %
+% ----------------------------------------------------------------------- %
+
+% Author: James Dunne, Chris Dembia, Tom Uchida,
+% Ajay Seth, Ayman Habib, Shrinidhi K. Lakshmikanth, Jen Hicks.
+
+% Import OpenSim Libraries
+import org.opensim.modeling.*
+
+% open dialog boxes to select the model
+[filename, pathname] = uigetfile('*.osim', 'Select an OpenSim Model File');
+
+% get the model path
+modelFilePath = fullfile(pathname,filename);
+
+% Generate an instance of the model
+model = Model(modelFilePath);
+
+% Get the number of coordinates  and a handle to the coordainte set
+nCoord = model.getCoordinateSet.getSize();
+coordSet = model.getCoordinateSet();
+
+% Create some empty vec3's for later.
+massCenter = Vec3();
+axisValues = Vec3();
+
+% Create an empty Force set
+forceSet = ForceSet();
+
+% Create the underlying computational System and return a handle to the State
+state = model.initSystem();
+
+optimalForce = 1;
+
+%% Start going through the coordinates, creating an actuator for each
+for iCoord = 0 : nCoord - 1
+
+    % get a reference to the current coordinate
+    coordinate = coordSet.get(iCoord);
+    % If the coodinate is constrained, don't add an actuator
+    if coordinate.isConstrained(state)
+        continue
+    end
+
+    % get the joint, parent and child names for the coordiante
+    joint = coordinate.getJoint();
+    parentName = joint.getParentFrame().getName();
+    childName = joint.getChildFrame().getName();
+
+    % If the coordinates parent body is connected to ground, we need to
+    % add residual actuators (torque or point).
+    if strmatch(parentName, model.getGround.getName() )
+
+        % if the joint type is custom or free
+        if strcmp(joint.getConcreteClassName(), 'CustomJoint') | strcmp(joint.getConcreteClassName(), 'FreeJoint')
+               % get the coordainte motion type
+               motion = char(coordinate.getMotionType);
+               % to get the axis value for the coordiante, we need to drill
+               % down into the coordinate transform axis
+               eval(['concreteJoint = ' char(joint.getConcreteClassName()) '.safeDownCast(joint);'])
+               sptr = concreteJoint.getSpatialTransform;
+               for ip = 0 : 5
+                  if strcmp(char(sptr.getCoordinateNames().get(ip)), char(coordinate.getName))
+                        sptr.getTransformAxis(ip).getAxis(axisValues);
+                        break
+                  end
+               end
+
+
+               % make a torque actuator if a rotational coordinate
+               if strcmp(motion, 'Rotational')
+                   newActuator = TorqueActuator(joint.getParentFrame(),...
+                                         joint.getParentFrame(),...
+                                         axisValues,...
+                                         1);
```

Comment:
Prefer `true` rather than `1` so it's clear that the last argument is a `bool`. A comment would be helpful, too (though, since the last argument defaults to `true` anyway, you could just omit it).